config/routes.rb

```diff
@@ -1,6 +1,14 @@
 Rails.application.routes.draw do
+  get 'users/index'
+  get 'users/show'
+  get 'posts/index'
+  get 'posts/show'
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
   # Defines the root path route ("/")
   # root "articles#index"
+  resources :users, only: [:index, :show] do
+    resources :posts, only: [:index, :show]
+  end
+  root "users#index"
```

Comment:
- [ ]  You did an amazing job to implement the routes, but you can still improve it. You can delete four lines of the basic routes `get`, because you don't need them now. just use the resources method is it helps us to define nested routes for the users and posts resources.

Reply:
Fixed, Thank you very much.